internal/service/cognitoidp/user_pool.go

```diff
@@ -1099,6 +1130,22 @@ func resourceUserPoolUpdate(d *schema.ResourceData, meta interface{}) error {
 			params.SmsConfiguration = expandSMSConfiguration(v.([]interface{}))
 		}
 
+		if v, ok := d.GetOk("user_attribute_update_settings"); ok {
+			configs := v.([]interface{})
+			config, ok := configs[0].(map[string]interface{})
+
+			if ok && config != nil {
+				params.UserAttributeUpdateSettings = expandUserPoolUserAttributeUpdateSettings(config)
+			}
+		}
+		if d.HasChange("user_attribute_update_settings") && params.UserAttributeUpdateSettings == nil {
```

Comment:
This conditional is attempting to address the situation where `user_attribute_update_settings` _was_ set, but is removed. 

In that situation `d.GetOk("user_attribute_update_settings")` returns a false `ok`, and `expandUserPoolUserAttributeUpdateSettings` isn't called. This leaves `params.UserAttributeUpdateSettings` as the nil zero value and the existing setting unmodified. `terraform apply` will return successfully, but won't actually make the update the user intends, and subsequent `plans` will always show a diff.

Reply:
I didn't see any other attributes handled this way, so I wasn't sure on the best approach. A couple other ideas that might make this more explicit:
1. Instead of the nil value, use the empty array as the default prior the the `d.GetOk` condition:
```go
		params.UserAttributeUpdateSettings = &cognitoidentityprovider.UserAttributeUpdateSettingsType{
			AttributesRequireVerificationBeforeUpdate: []*string{},
		}
		if v, ok := d.GetOk("user_attribute_update_settings"); ok {
			// ...
		}
```
2. Add an `else` on the ok check, making it more obvious the setting no longer exists
```go
		if v, ok := d.GetOk("user_attribute_update_settings"); ok {
			// ...
		} else {
		        params.UserAttributeUpdateSettings = &cognitoidentityprovider.UserAttributeUpdateSettingsType{
		    	    AttributesRequireVerificationBeforeUpdate: []*string{},
		        }
                }
```

Do either of these seem better, or maybe something simpler I'm missing?